src/feedClass.ts

```diff
@@ -0,0 +1,281 @@
+    gunUser = this.gun.user();
+    feeds = null;
+
+    feedItemDataFields = [
+        'rom',
+        'amount',
+        'code',
+        'reason',
+        'counterPartyDisplayName',
+        'paymentLink',
+        'otplStatus',
+        'customName',
+        'subtitle',
+    ];
+    feedInfoFields = [
+        'name',
+        'id',
+        'owner',
+    ];
+    feedItemFields = [
+        'id',
+        'type',
+        'status',
+        'createdDate',
+        'data',
+    ];
+
+    /**
+     * Auth in gun and return all feeds
+     *
+     * @param {string} username
+     * @param {string} password
+     *
+     * @return Feeds | boolean
+     */
+    async login(username: string, password: string): Promise<Feeds | boolean> {
+        logger.info('class login');
+        const user = await this.gunAuthInitFeed(username, password);
+        return (user)
+    }
+
+    /**
+     * add item to feed
+     * @param {string} feedInfoName
+     * @param {FeedItem} feedItem
+     */
+    async addItem(feedInfoName: string, feedItem: FeedItem): Promise<boolean> {
+        const date = new Date();
+        const day = `${date.toISOString().slice(0, 10)}`;
+
+        await this.feeds.get(feedInfoName).get('items').get(day).put(feedItem);
+        return true
+    }
+
+    /**
+     * Add private item to feed
+     * @param {string} feedInfoName
+     * @param {FeedItem} feedItem
+     * @param {string} privateKey
+     */
+    async addPrivateItem(feedInfoName: string, feedItem: FeedItem, privateKey: string): Promise<boolean> {
+        const date = new Date();
+        const day = `${date.toISOString().slice(0, 10)}`;
+
+        await this.feeds.get(feedInfoName).get('privateItems').get(privateKey).get(day).put(feedItem);
+        return true
+    }
+
+    /**
+     * Create new feed
+     *
+     * @param {Feed} feed
+     */
+    async createFeed(feed: Feed): Promise<Feed | any> {
+        logger.info('class createFeed');
+
+        feed.info.owner = await this.getOwnerSoul();
+        const gunNewFeed = this.feeds.get(feed.info.name).put(feed);
+        const id = await this.getSoul(gunNewFeed);
+        await gunNewFeed.get('info').get('id').put(id);
+        return gunNewFeed;
+    }
+
+
+    /**
+     * Return user soul
+     */
+    async getOwnerSoul(): Promise<string> {
+        return await this.getSoul(this.gunUser)
+    }
+
+
+    /**
+     * Return soul gun object
+     * @param {Gun} gunInstance
+     */
+    async getSoul(gunInstance: Gun): Promise<string> {
+        return (await gunInstance)['_']['#']
+    }
+
+    /**
+     * Get feed by id
+     *
+     * @param {string} feedInfoName
+     */
+    async getFeed(feedInfoName: string): Promise<Feed | any> {
+        logger.info('class getFeed', {feedInfoName});
+        let result: any = {
+            info: {},
+            items: {},
+            privateItems: {}
+        };
+
+        try {
+            const info = await this.feeds.get(feedInfoName).get('info').then()
```

Comment:
there is a .path() shortener allows to avoid log .get() chains
https://gun.eco/docs/API#-a-name-path-a-gun-path-key-